feat(Tree): add badge icon to tree node item

leko91 opened the pull request:
As discussed somewhat in the [Slack thread](https://frontify.slack.com/archives/C02PSJTPA3D/p1650372981299969).

I did the styling as it is in the [design for the Navigation Manager](https://www.figma.com/file/f37w3IPvu5wBpo0GakMNm0/Navigation?node-id=985%3A467784).

julianiff commented on src/components/Tree/Tree.tsx:
```diff
@@ -15,6 +15,7 @@ export interface TreeFlatListItem {
     label?: string;
     value?: string;
     actions?: React.ReactNode[];
+    badge?: ReactElement<IconProps>;
```
Would we be able to restrict a bit more what we put into the Badge? Because its called Badge, i think it would make sense to restrict to Icons/Badge, and also what you would need to fulfill your requirements.

julianiff commented on src/components/Tree/Node.tsx:
```diff
@@ -137,7 +137,19 @@ export const Node = ({
+                                            selected && "tw-bg-transparent",
+                                        ])}
+                                    >
+                                        <span>{badge}</span>
```
```suggestion
                                       {badge}
```

Do we need a span here? if not we could leave it out.

julianiff commented on src/components/Tree/Node.tsx:
```diff
@@ -137,7 +137,19 @@ export const Node = ({
-                            <span data-test-id="node-link-name">{name}</span>
+                            <span className="tw-flex tw-items-center" data-test-id="node-link-name">
+                                {name}
+                                {badge && (
+                                    <div
+                                        className={merge([
+                                            "tw-flex tw-justify-center tw-items-center tw-w-8 tw-h-5 tw-ml-2 tw-bg-box-neutral tw-rounded-full",
+                                            selected && "tw-bg-transparent",
+                                        ])}
+                                    >
+                                        <span>{badge}</span>
+                                    </div>
+                                )}
```
What do you think, would it make sense if we refactor this new addition to the Node into a separate function, smth like const insertBadge/addBadge that suggests that a Badge gets included.

julianiff commented on src/components/Tree/utils/mocks.tsx:
```diff
@@ -141,6 +141,7 @@ export const mockNodesFlat: DraggableItem<TreeFlatListItem>[] = [
         icon: <IconFolder size={IconSize.Size16} />,
         value: "https://weare.frontify.com/document/923#/test",
         sort: null,
+        badge: <IconSmileysPeople size={IconSize.Size16} />,
```
Looove the smily :D 
We also have the Badge Component, maybe we could include also the Badge in another Node to also show this showcase.

julianiff reviewed (commented):
Looks great thank you for the PR 👍. I added some small comments from my side.

leko91 commented:
@julianiff
I've updated the PR based on your comments. Also, I've added a bit more styling since Badge and our Icon differentiate a bit. Maybe that's something that the design team could fix, so we always use only the badge element, if they agree we could remove the Icon use case.

julianiff commented on src/components/Tree/Node.tsx:
```diff
@@ -85,6 +85,22 @@ export const Node = ({
     };
 
+    const insertBadge = () => {
+        console.log(badge);
```
```suggestion
```

julianiff reviewed (approved):
Looks great. There is just a console.log to remove.

julianiff commented:
Ah if possible, could we add also a test that checks if the badge is rendered correctly? That could avoid later regression bugs if someone would change the badge.

leko91 commented:
> Ah if possible, could we add also a test that checks if the badge is rendered correctly? That could avoid later regression bugs if someone would change the badge.

@julianiff 
Added the tests now 😄

thisguyscode commented:
@julianiff I thought we would use a proper Badge component. Is there a limitation to the Badge that made that difficult/impossible?
